.travis.yml

```diff
@@ -1,23 +1,41 @@
 language: cpp
+matrix:
+  include:
+    - os: linux
+      dist: xenial
+      addons:
+        apt:
+          sources:
+            - ubuntu-toolchain-r-test
+          packages:
+            - g++-7
+      env:
+        - MATRIX_EVAL="CXX=g++-7"
 
-image: xenial
+    - os: linux
+      dist: xenial
+      addons:
+        apt:
+          sources:
+            - ubuntu-toolchain-r-test
+          packages:
+            - g++-8
+      env:
+         - MATRIX_EVAL="CXX=g++-8"
 
-compiler:
-  - clang
-  - gcc
+    - os: osx
+      osx_image: xcode10.1
 
+    - os: osx
+      osx_image: xcode10.2
 before_install:
-  - sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
-  - sudo apt-get update -qq
-  - sudo apt-get install g++-8 libstdc++-8-dev
-  - sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-8 90
-  - sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-8 90
-
+  - eval "${MATRIX_EVAL}"
```

Comment:
instead of using `eval` here,  why not use
```yaml
env:
  - CXX="g++-8"
```
directly?

Reply:
Yes, it is the recommended way!

Reply:
Travis CI overwrites it. At least `eval "${MATRIX_EVAL}"` works fine!